### DISCUSSION TITLE
Tutorial for integrating authentication with Nylo

### DISCUSSION THREAD

**Discussion by WickedBrat** [nylo-core/nylo#42] [Q&A] [answered]
Hey guys,

I'm fairly new to Nylo and flutter and was looking for a way to add protected routes, maintain state after login etc, stuff like that.
I'd really appreciate a heads up on any tutorials on integrating the authentication with Nylo/Flutter and RouteGaurds. 

I see that router.route accepts `routeGaurds` parameter but not sure how to use it.

TIA

**Comment by agordn52** [comment]
Hi @WickedBrat,

Hope all is well.

I'll have to get back to you on this, keep you posted.

**Reply by agordn52** [reply, marked as answer]
Here are the docs for using route guards in Nylo https://nylo.dev/docs/5.x/router#route-guards
